ta/pkcs11/src/processing.c

```diff
@@ -647,14 +673,25 @@ enum pkcs11_rc entry_processing_key(struct pkcs11_client *client,
 		 * key. Hence pass function as PKCS11_FUNCTION_ENCRYPT
 		 * to init_symm_operation()
 		 */
-		rc = init_symm_operation(session, PKCS11_FUNCTION_ENCRYPT,
-					 proc_params, parent);
+		rc = init_symm_operation(session, operation, proc_params,
+					 parent);
 		if (rc)
 			goto out;
 
 		session->processing->mecha_type = proc_params->id;
 
-		rc = derive_key_by_symm_enc(session, &out_buf, &out_size);
+		switch (function) {
+		case PKCS11_FUNCTION_DERIVE:
+			rc = derive_key_by_symm_enc(session, &out_buf,
+						    &out_size);
+			break;
+		case PKCS11_FUNCTION_UNWRAP:
+			rc = unwrap_key_by_symm(session, in_buf, in_size,
+						&out_buf, &out_size);
+			break;
+		default:
+			rc = PKCS11_CKR_ARGUMENTS_BAD;
```

Comment:
prefer `rc = PKCS11_CKR_GENERAL_ERROR;`
Maybe also `assert(0)` or even `TEE_Panic(PKCS11_CKR_GENERAL_ERROR)`.

Reply:
Changed and also added a TEE_Panic(function);

Reply:
Changed